File: src/MaksIT.WebUI/src/models/letsEncryptServer/apiKeys/CreateApiKeyRequest.ts
import { object, Schema, string } from 'zod'
import { RequestModelBase } from '../../RequestModelBase'

export interface CreateApiKeyRequest extends RequestModelBase {
  description?: string
  expiresAt?: string
}

export const CreateApiKeyRequestSchema: Schema<CreateApiKeyRequest> = object({
  description: string().optional(),
  expiresAt: string().optional(),
})

File: src/MaksIT.CertsUI/Models/APIKeys/CreateApiKeyRequest.cs
using MaksIT.Core.Abstractions.Webapi;
using MaksIT.CertsUI.Engine;


namespace MaksIT.CertsUI.Models.APIKeys;


public class CreateApiKeyRequest : RequestModelBase {
  public string? Description { get; set; }
  public DateTime? ExpiresAt { get; set; }
  public bool IsGlobalAdmin { get; set; }
  public List<CreateApiKeyEntityScopeRequest>? EntityScopes { get; set; }
}
